### PR TITLE
Refactor/router single movie

### PR DESCRIPTION
**Pull request by saraho1123**
## Description

Use router to change between container and single movie pages with updated urls.

## Motivation and Context

The goal of this iteration is to use Router in our project. We had to refactor and move several components in order to get Router to work.  

## How Has This Been Tested?

We have not updated our tests yet. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Tickets:
#32 #34

### CODE REVIEW

File: src/MovieCard/MovieCard.css [hunk 1]
```diff
@@ -7,6 +7,8 @@
   margin: 1em;
   border: rgb(92, 92, 231) 1em solid;
   border-radius: .5em;
+  text-decoration: none;
```
Comment (saraho1123) [inline src/MovieCard/MovieCard.css:10]: I added some very small CSS changes!

File: src/SingleMovie/SingleMovie.js [hunk 2]
```diff
@@ -1,26 +1,50 @@
...
+    this.state= {
+      movieDetails: [],
+    }
+  }
+
+  componentDidMount = () => {
+    this.getSingleMovie(this.props.id)
+
+  }
+
```
Comment (saraho1123) [inline src/SingleMovie/SingleMovie.js:18]: Glad we consolidated these methods a bit!

File: src/index.js [hunk 3]
```diff
@@ -3,13 +3,15 @@ import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App/App';
 import reportWebVitals from './reportWebVitals';
+import { BrowserRouter } from 'react-router-dom';
 
-ReactDOM.render(
-  <React.StrictMode>
+const router = (
```
Comment (saraho1123) [inline src/index.js:8]: Router is challenging, but kind of fun!

**Review by saraho1123 — COMMENTED** [review]
Thanks for getting John Adams help on Router!

File: src/MovieCard/MovieCard.css (hunk 1 above)
Comment (chadrick-d-dev) [inline src/MovieCard/MovieCard.css:10]: This was an interesting discovery for removing link underlines.

File: src/MovieCard/MovieCard.js [hunk 4]
```diff
@@ -1,16 +1,19 @@
...
+const MovieCard = ( {id, image, title, rating }) => {
   return (
-    <section className="movie-card" onClick={() => {getMovieDetails(id)} }>
-      <div className="image-holder">
-        <img className="card-image" src={image} alt={title}/>
-      </div>
-      <h3 className="card-title">{title}</h3>
-      <h3 className="rating">Rating: {Math.round( rating * 10 ) / 10}</h3>
-    </section>
+    <Link to={`/${id}`} className="movie-card-link">
```
Comment (chadrick-d-dev) [inline src/MovieCard/MovieCard.js:7]: MovieCard became the link to SingleMovie! Pretty Cool

File: src/SingleMovie/SingleMovie.js [hunk 5]
```diff
@@ -1,26 +1,50 @@
...
-        <h3 className="card-title">{movieDetails.title}</h3>
-        <h3 className="rating">Rating: {Math.round( movieDetails.average_rating * 10 ) / 10}</h3>
-        <h3 className="release-date">Release Date: {movieDetails.release_date}</h3>
-        <h3 className="overview">Overview: {movieDetails.overview}</h3>
-        <h3 className="genres">Genres: {movieDetails.genres}</h3>
-        <h3 className="budget">Budget: ${movieDetails.budget}.00</h3>
-        <h3 className="revenue">Revenue: ${movieDetails.revenue}.00</h3>
-        <h3 className="runTime">Run-Time: {movieDetails.runtime} minutes</h3>
-        <h3 className="tagline">Tagline: {movieDetails.tagline}</h3>
+class SingleMovie extends Component {
```
Comment (chadrick-d-dev) [inline src/SingleMovie/SingleMovie.js:6]: I'm glad we changed this to a class! No more infinity loop!

**Review by chadrick-d-dev — APPROVED** [review]

**Pull request merged**